ecommerce/extensions/offer/utils.py

```
@@ -99,3 +104,49 @@ def render_email_confirmation_if_required(request, offer, product):
                 'user_email': request.user and request.user.email,
             }
         )
+
+
+def send_assigned_offer_email(
+        template,
+        offer_assignment_id,
+        learner_email,
+        code,
+        enrollment_url,
+        redemptions_remaining,
+        code_expiration_date):
+    """
+    Arguments:
+        *template*
+            The email template with placeholders that will receive the following tokens
+        *offer_assignment_id*
+            Primary key of the entry in the offer_assignment model.
+        *learner_email*
+            Email of the customer who will receive the code.
+        *code*
+            Code for the user.
+        *enrollment_url*
+            URL for the user.
+        *redemptions_remaining*
+            Number of times the code can be redeemed.
+        *code_expiration_date*
+            Date till code is valid.
+
+    Returns:
+         True when successful or False in case of any exception
+    """
+
+    email_subject = settings.OFFER_ASSIGNMENT_EMAIL_DEFAULT_SUBJECT
+    try:
+        email_body = template.format(
+            REDEMPTIONS_REMAINING=redemptions_remaining,
+            USER_EMAIL=learner_email,
+            ENROLLMENT_URL=enrollment_url,
+            CODE=code,
+            EXPIRATION_DATE=code_expiration_date
+        )
+        send_offer_assignment_email.delay(learner_email, offer_assignment_id, email_subject, email_body)
+    except Exception as exc:  # pylint: disable=broad-except
```

in my PR for revoke i moved this to the serializer so we could return the exception message to the caller (https://github.com/edx/ecommerce/pull/2087/files#diff-852a6b45e4fa1255461cf3fa6e5ceda7R1202), which i think is preferable. what do you think? i'd like these email util methods to be consistent with one another

---

Changed as suggested.

---

@brittneyexline I have moved the exception handling to the serializer. Do you think the 'detail' key value as used in CouponCodeRevokeBulkSerializer is required for the CouponCodeAssignmentSerializer as well?

---

For email, an exception can be raised while we are initiating the ecommerce-worker celery task and also within ecommerce worker. In both cases, the offer assignment status remains at the initial default status of 'EMAIL_PENDING'. Hence, if we send back the 'detail', it would only cover the errors raised while initiating the ecommerce-worker celery task.